api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -435,6 +435,19 @@ model NodePoolPlatformProfile {
   @visibility(Lifecycle.Read, Lifecycle.Create)
   diskStorageAccountType?: DiskStorageAccountType = "Premium_LRS";
 
+  /** The ID of the DiskEncryptionSet resource to use to encrypt the OS disks for the VMs.
+   * Can be used with either platform (Azure) managed, or customer managed encryption keys.
+   * This needs to exist in the same subscription id listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.SubscriptionID.
+   * DiskEncryptionSetID should also exist in a resource group under the same subscription id and the same location
+   * listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.Location.
+   *
+   * Details on how to create a Disk Encryption Set can be found here:
+   * https://learn.microsoft.com/en-us/azure/virtual-machines/disks-enable-customer-managed-keys-portal#set-up-your-disk-encryption-set
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  @maxLength(285)
+  diskEncryptionSetId?: url;
```

Comment:
I would make clear that this is for the *OS Disk* as part of the name. Unless we are intentionally designing this same one be used for potential future data disks

Reply:
All the other attributes for attributes are just disk so I'd rather keep them all as 'disk***' or nest them all under some 'NodePoolOSDiskSpec' object so the context is clear (this is what [hypershift does](https://github.com/openshift/hypershift/blob/main/api/hypershift/v1beta1/azure.go#L273))... but maybe we've already gone too far and this would be a breaking change already?? @mbarnes-azure any thoughts on this?

Reply:
I think renaming or grouping all the `disk*` fields at this point would have to be a BU decision.  Maybe bring it up in the next API meeting?

My suggestion for now is just make sure all the doc comments for these fields say "OS disk".  The comment for `diskStorageAccountType` might need clarified.

Reply:
I would prefer having them as os_disk_* or within an os disk section. If we leave them with generic name and in the future we add support for data disks we will end up with generic attribute names and then data_* names which is not ideal, because we will probably want to have differentiated configuration for them.

Reply:
We decided to nest things under OSDisk